server/src/test/java/org/apache/iotdb/db/integration/IoTDBSimpleQueryTest.java

```diff
@@ -41,6 +44,62 @@ public void tearDown() throws Exception {
     EnvironmentUtils.cleanEnv();
   }
 
+  @Test
+  public void testEmptyDataSet() throws SQLException, ClassNotFoundException {
+    Class.forName(Config.JDBC_DRIVER_NAME);
+    try(Connection connection = DriverManager
+        .getConnection(Config.IOTDB_URL_PREFIX + "127.0.0.1:6667/", "root", "root");
+        Statement statement = connection.createStatement()){
+
+      ResultSet resultSet = statement.executeQuery("select * from root");
+      // has an empty time column
+      Assert.assertEquals(1, resultSet.getMetaData().getColumnCount());
+      while(resultSet.next()) {
+        fail();
+      }
+
+      resultSet = statement.executeQuery(
+          "select count(*) from root where time >= 1 and time <= 100 group by ([0, 100), 20ms, 20ms)");
+      // has an empty time column
+      Assert.assertEquals(1, resultSet.getMetaData().getColumnCount());
+      while (resultSet.next()) {
+        fail();
+      }
+
+      resultSet = statement.executeQuery("select count(*) from root");
+      // has no column
+      Assert.assertEquals(0, resultSet.getMetaData().getColumnCount());
+      while(resultSet.next()) {
+        fail();
+      }
+
+      resultSet = statement.executeQuery("select * from root align by device");
+      // has time and device columns
+      Assert.assertEquals(2, resultSet.getMetaData().getColumnCount());
+      while(resultSet.next()) {
+        fail();
+      }
+
+      resultSet = statement.executeQuery("select count(*) from root align by device");
+      // has device column
+      Assert.assertEquals(1, resultSet.getMetaData().getColumnCount());
+      while(resultSet.next()) {
+        fail();
+      }
+
+      resultSet = statement.executeQuery(
+          "select count(*) from root where time >= 1 and time <= 100 "
+              + "group by ([0, 100), 20ms, 20ms) align by device");
+      // has an empty time column
+      Assert.assertEquals(2, resultSet.getMetaData().getColumnCount());
```

Comment:
The comment here should be has an empty time column and device column

Reply:
fixed